File: common/src/main/java/com/hartwig/actin/molecular/datamodel/driver/Variant.java
package com.hartwig.actin.molecular.datamodel.driver;

import java.util.Set;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class Variant implements Driver, GeneAlteration {

    @NotNull
    public abstract VariantType type();

    public abstract double variantCopyNumber();

    public abstract double totalCopyNumber();

    public abstract boolean isBiallelic();

    public abstract boolean isHotspot();

    public abstract double clonalLikelihood();

    @NotNull
    public abstract TranscriptImpact canonicalImpact();

    @NotNull
    public abstract Set<TranscriptImpact> otherImpacts();

    //public abstract boolean isReportable();
}

File: common/src/main/java/com/hartwig/actin/molecular/datamodel/driver/Variant.kt
package com.hartwig.actin.molecular.datamodel.driver

import org.immutables.value.Value
import org.jetbrains.annotations.NotNull
import org.jetbrains.annotations.Nullable

@Value.Immutable
@Value.Style(passAnnotations = [NotNull::class, Nullable::class])
abstract class Variant : Driver, GeneAlteration {
    abstract fun type(): VariantType
    abstract fun variantCopyNumber(): Double
    abstract fun totalCopyNumber(): Double

    @JvmField
    abstract val isBiallelic: Boolean

    @JvmField
    abstract val isHotspot: Boolean
    abstract fun clonalLikelihood(): Double
    abstract fun phaseGroups(): Set<Int?>?
    abstract fun canonicalImpact(): TranscriptImpact
    abstract fun otherImpacts(): Set<TranscriptImpact?>
}
